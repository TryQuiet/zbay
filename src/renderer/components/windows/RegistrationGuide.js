import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { shell } from 'electron'

import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
import { Grid, Button } from '@material-ui/core'

const reqSvgs = require && require.context('../../static/images/registrationGuide/', true, /\.svg$/)

const styles = theme => ({
  root: {
    width: '100%',
    height: '100vh',
    backgroundColor: theme.palette.colors.white
  },
  main: {
    width: 800,
    height: 500
  },
  dot: {
    height: 8,
    width: 8,
    backgroundColor: theme.palette.colors.zbayBlue,
    borderRadius: '50%',
    opacity: 0.2,
    marginLeft: 6,
    display: 'inline-block'
  },
  buttonBack: {
    width: 120,
    height: 60,
    fontSize: 16,
    lineHeight: '19px',
    fontStyle: 'normal',
    fontWeight: 'normal',
    textTransform: 'none',
    border: `1px solid ${theme.palette.colors.zbayBlue}`,
    borderRadius: 4,
    textAlign: 'center',
    color: theme.palette.colors.zbayBlue,
    backgroundColor: theme.palette.colors.white,
    '&:hover': {
      backgroundColor: theme.palette.colors.white
    }
  },
  buttonNext: {
    width: 120,
    height: 60,
    fontSize: 16,
    lineHeight: '19px',
    fontStyle: 'normal',
    fontWeight: 'normal',
    textTransform: 'none',
    border: `1px solid ${theme.palette.colors.zbayBlue}`,
    borderRadius: 4,
    textAlign: 'center',
    color: theme.palette.colors.white,
    backgroundColor: theme.palette.colors.zbayBlue,
    '&:hover': {
      backgroundColor: theme.palette.colors.zbayBlue
    }
  },
  navigationContainer: {
    width: '100%',
    padding: '0px 45px'
  },
  contentContainer: {
    padding: '0px 48px',
    marginBottom: 28
  },
  graphicBox: {
    width: 216,
    height: 356
  },
  textBox: {
    width: 456,
    height: 356
  },
  active: {
    opacity: 1
  },
  sentence: {
    marginTop: 24
  },
  wideButton: {
    width: 237
  },
  alignDots: {
    marginRight: 46
  },
  linkBlue: {
    cursor: 'pointer',
    color: theme.palette.colors.linkBlue
  }
})

const ContentWithRedirect = ({ classes }) => (
  <Fragment>
    <Grid item>
      <Typography variant={'h3'}>And now... some waiting!</Typography>
      <Typography className={classes.sentence} variant={'body2'}>
      Our tour has concluded! Now, we just have to wait for some blockchain files to download. Perhaps leave your computer and go have a sandwich.
      </Typography>
      <Typography className={classes.sentence} variant={'body2'}>
    If you'd like to learn more about the values and vision behind the Zbay project, there's <span onClick={() => shell.openExternal('https://www.zbay.app/#why')} className={classes.linkBlue}>a great (and long) essay on our site.</span>
      </Typography>
      <Typography className={classes.sentence} variant={'body2'}>
    Or you can learn more about <span onClick={() => shell.openExternal('https://zcash.readthedocs.io/en/latest/')} className={classes.linkBlue}>Zcash</span> the technology that Zbay builds on.
      </Typography>
    </Grid>
  </Fragment>
)

export const RegistraionGuide = ({ classes, content, currentSlide, prevSlide, nextSlide, setStoryStatus }) => {
  return (
    <Grid container className={classes.root} justify={'center'} alignItems={'center'} alignContent={'center'}>
      <Grid container item className={classes.main}>
        <Grid container item className={classes.contentContainer} justify={'space-between'}>
          <Grid container item className={classes.graphicBox} alignItems={'center'}>
            <img src={reqSvgs(content[currentSlide].fileName)} />
          </Grid>
          <Grid container item className={classes.textBox} alignItems={'center'}>
            {currentSlide !== 10
              ? <Fragment>
                <Grid item>
                  <Typography variant={'h3'}>{content[currentSlide].title}</Typography>
                  {content[currentSlide].sentences.map(sentence => <Typography className={classes.sentence} variant={'body2'}>{sentence}</Typography>)}
                </Grid>
              </Fragment> : <ContentWithRedirect classes={classes} />}
          </Grid>
        </Grid>
        <Grid container item className={classes.navigationContainer} justify={'center'} alignItems={'center'} wrap={'nowrap'}>
          <Grid item>
            <Button onClick={prevSlide} className={classes.buttonBack}>Back</Button>
          </Grid>
          <Grid container item justify={currentSlide === 10 ? 'flex-end' : 'center'} className={classNames(null, { [classes.alignDots]: currentSlide === 10 })}>
            {content.map((_, index) => <Grid key={index} item><span className={classNames(classes.dot, {
              [classes.active]: index === currentSlide
            })} /></Grid>)}
          </Grid>
          <Grid item>
            <Button onClick={currentSlide !== 10 ? nextSlide : setStoryStatus} className={classNames(classes.buttonNext, { [classes.wideButton]: currentSlide === 10 })}>
              {currentSlide !== 10 ? 'Next' : 'Show download progress'}
            </Button>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  )
}

RegistraionGuide.propTypes = {
  classes: PropTypes.object.isRequired,
  content: PropTypes.array.isRequired,
  currentSlide: PropTypes.number.isRequired,
  prevSlide: PropTypes.func.isRequired,
  nextSlide: PropTypes.func.isRequired,
  setStoryStatus: PropTypes.func.isRequired
}

RegistraionGuide.defaultProps = {
}

export default withStyles(styles)(RegistraionGuide)
