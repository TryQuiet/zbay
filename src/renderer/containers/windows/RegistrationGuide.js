import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import RegistrationGuideComponent from '../../components/windows/RegistrationGuide'
import text from '../../static/text/registrationGuide'
import electronStore from '../../../shared/electronStore'
import nodeHandlers from '../../../renderer/store/handlers/node'
import nodeSelector from '../../../renderer/store/selectors/node'

export const mapStateToProps = state => ({
  getCurrentSlide: nodeSelector.currentSlide(state)
})

export const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      setGuideStatus: nodeHandlers.actions.setGuideStatus,
      setNextSlide: nodeHandlers.actions.setNextSlide,
      setPrevSlide: nodeHandlers.actions.setPrevSlide
    },
    dispatch
  )
}

export const RegistrationGuide = ({ setGuideStatus, getCurrentSlide, setNextSlide, setPrevSlide }) => {
  const setStoryStatus = () => {
    setGuideStatus(true)
    electronStore.set('storyStatus', true)
  }
  return <RegistrationGuideComponent setStoryStatus={setStoryStatus} content={text} currentSlide={getCurrentSlide} prevSlide={setPrevSlide} nextSlide={setNextSlide} />
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RegistrationGuide)
