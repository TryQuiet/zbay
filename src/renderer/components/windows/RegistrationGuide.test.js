import React from 'react'
import { shallow } from 'enzyme'
import { RegistraionGuide } from './RegistrationGuide'
import { mockClasses } from '../../../shared/testing/mocks'

describe('RegistraionGuide', () => {
  it('renders component', () => {
    const result = shallow(
      <RegistraionGuide
        classes={mockClasses}
        isGuideCompleted
        content={[{
          slideId: 9,
          title: 'Buying & selling',
          sentences: ['You can buy and sell things on Zbay, privately.', 'Zbay lets you send a seller your order information (quantity, size, etc.) and your shipping address (if necessary) with funds in an encrypted message.',
            'With Bitcoin, third parties can see who paid who, but Zcash makes this much more difficult, if not impossible, protecting your privacy. Someday Zbay will have escrow payments with dispute resolution and seller ratings. Right now, channel owners can use moderation to keep out scam sellers.'
          ],
          fileName: './buying.svg'
        },
        {
          slideId: 10,
          title: 'And now... some waiting!',
          sentences: ['Our tour has concluded! Now, we just have to wait for some blockchain files to download. Perhaps leave your computer and go have a sandwich.',
            'If you\'d like to learn more about the values and vision behind the Zbay project, there\'s a great (and long) essay on our site.', 'Or you can learn more about Zcash the technology that Zbay builds on.'
          ],
          fileName: './hourglass.svg'
        }]}
        currentSlide={1}
        prevSlide={jest.fn()}
        nextSlide={jest.fn()}
        setStoryStatus={jest.fn()}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
