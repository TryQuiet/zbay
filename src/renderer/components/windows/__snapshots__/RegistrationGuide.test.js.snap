// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistraionGuide renders component 1`] = `
<WithStyles(ForwardRef(Grid))
  alignContent="center"
  alignItems="center"
  className="root-mock"
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    alignContent="center"
    alignItems="center"
    className="wrapper-mock"
    container={true}
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      className="contentContainer-mock"
      container={true}
      item={true}
      justify="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        className="graphicBox-mock"
        container={true}
        item={true}
      >
        <img
          src={
            Object {
              "default": "test-file-stub",
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        className="textBox-mock"
        container={true}
        item={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Typography))
            variant="h3"
          >
            And now... some waiting!
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            className="sentence-mock"
            key="0"
            variant="body2"
          >
            Our tour has concluded! Now, we just have to wait for some blockchain files to download. Perhaps leave your computer and go have a sandwich.
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            className="sentence-mock"
            key="1"
            variant="body2"
          >
            If you'd like to learn more about the values and vision behind the Zbay project, there's a great (and long) essay on our site.
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            className="sentence-mock"
            key="2"
            variant="body2"
          >
            Or you can learn more about Zcash the technology that Zbay builds on.
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      className="navigationContainer-mock"
      container={true}
      item={true}
      justify="center"
      wrap="nowrap"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Button))
          className="buttonBack-mock"
          onClick={[MockFunction]}
        >
          Back
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className=""
        container={true}
        item={true}
        justify="center"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          key="0"
        >
          <span
            className="dot-mock"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          key="1"
        >
          <span
            className="dot-mock active-mock"
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Button))
          className="buttonNext-mock"
          onClick={[MockFunction]}
        >
          Next
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
